### ISSUE TITLE
Problems to activate the SASjs extension in VS Code

### ISSUE THREAD

**Issue by grandgrue** [sasjs/vscode-extension#388]
I tried several installations of the extension.
In the VS Code developer console I get the following error message when opening a file with .sas extension: mainThreadExtensionService.ts:111 Activating extension 'SASjs.sasjs-for-vscode' failed: Cannot read properties of undefined (reading '0').

If i click on any of the buttons, I get message boxes popping up. For example if I click on the "SASjs: Select Target" button, the message says: 
command 'sasjs-for-vscode.selectTarget' not found

Do you know of such installation problems (I tried v.1.28 and v1.27.3 from marketplace.visualstudio.com and open-vsx.org)
Any help or hints are much appreciated,
Michael

**Comment by allanbowe** [comment]
Hi Michael - thanks for getting in touch.  Can you confirm that you are using the latest version of VS Code?  Are you running on Windows OS, if so, which one please?

It would be great to have a call to bottom this one out - https://sasapps.io/contact

/Allan

**Comment by grandgrue** [comment]
Thank you for your fast reply.
Yes, I am using the latest VS Code Version for Windows. Here are the details from "Help" - "About":
Version: 1.74.3 (system setup)
Commit: 97dec172d3256f8ca4bfb2143f3f76b503ca0534
Date: 2023-01-09T16:59:02.252Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No

**Comment by allanbowe** [comment]
We just tested with the following and it works for us:

![image](https://user-images.githubusercontent.com/4420615/215087033-4145f717-b096-4ded-a4b0-0b840b65e431.png)

Can you try re-installing?

Alternatively we'd love to jump on a quick call and see if we can help troubleshoot.

**Comment by allanbowe** [comment]
Thanks for your time to show us the issue(s)! Unfortunately we could not reproduce the execution one - for that we will need another session to see if your site definitely does have the `_user=` and `_password=` login option available (not disabled).

The issues we will fix as part of this ticket are:

- Popup to explain that you must be in a folder for the extension to work (as it searches for a local target in this folder)
- Improved text when there are no targets ("Please select from the list below")
- When no runner is registered, the log / output window includes a link to the relevant SAS code - https://cli.sasjs.io/auth/#sasjs-runner

**Comment by grandgrue** [comment]
I was able to solve the last open problem on my own:
In our IT infrastructure we are using proxy-servers for all outgoing communication. For most open source software we therefore have to set the proxy-credentials with envirionment variables:
`setx HTTP_PROXY http://{proxyuser}:{proxypassword}@{proxy-server}:8080`
But in this case, we are using an internal server, where no proxy shall be used. So once I added by SAS server to the exclude list, SASjs worked:
`setx NO_PROXY {sas-server}`
I will therefore closs this issue now. Thanks for your help!
